iam.tf

```diff
@@ -0,0 +1,42 @@
+resource "aws_iam_role" "this" {
+  name               = "${var.git}-state-machine-role"
+  assume_role_policy = data.aws_iam_policy_document.assume_role.json
+}
+
+data "aws_iam_policy_document" "assume_role" {
+  statement {
+    effect  = "Allow"
+    actions = ["sts:AssumeRole"]
+
+    principals {
+      identifiers = ["states.amazonaws.com"]
+      type        = "Service"
+    }
+  }
+}
+
+resource "aws_iam_policy" "this" {
+  name_prefix = var.git
+  policy      = data.aws_iam_policy_document.this.json
+}
+
+resource "aws_iam_role_policy_attachment" "this" {
+  policy_arn = aws_iam_policy.this.arn
+  role       = aws_iam_role.this.name
+}
+
+data "aws_iam_policy_document" "this" {
```

Comment:
i think we're going to need a way to add lambda permissions here

Reply:
ok, I'll take a look at adding lambda execute permissions

Reply:
also added xray policies in case tracing needs to be enabled.